### PR TITLE
Add Elasticsearch to list of docker containers

### LINKED ISSUE #3: Add support to deploy ElasticSearch
https://hub.docker.com/_/elasticsearch

Add support to deploy all releases of ElasticSearch for which the docker images are available..

### PR DESCRIPTION
**Pull request by abprash**
Closes #3

### CODE REVIEW

File: src/main/java/DockerImage.java [hunk 1]
```diff
@@ -45,7 +45,12 @@
             "latest",
             "8000:8000 8089:8089",
             "SPLUNK_START_ARGS=--accept-license;SPLUNK_PASSWORD=initialPassword12345!"
-    );
+    ),
+    ELASTICSEARCH_7_9_2(
```
Comment (abprash) [inline src/main/java/DockerImage.java:49]: ES doesn't have a generic latest tag.

**Review by the-c0d3br34k3r — CHANGES_REQUESTED** [review]

File: src/test/java/ElasticsearchDockerImageTest.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+public class ElasticsearchDockerImageTest {
+
+    private static final Logger logger = LogManager.getLogger(ElasticsearchDockerImageTest.class);
+    private static final RequestOptions COMMON_OPTIONS;
+    private static RestClient restClient = RestClient.builder(
+            new HttpHost("localhost", 9200, "http"),
+            new HttpHost("localhost", 9300, "http")).build();
+    static {
+        RequestOptions.Builder builder = RequestOptions.DEFAULT.toBuilder();
```
[the remark this answers is not in the document]
Comment (abprash) [reply src/test/java/ElasticsearchDockerImageTest.java:28]: Sure makes sense.

File: src/main/java/DockerImage.java [hunk 3]
```diff
@@ -45,7 +45,12 @@
...
+            "elasticsearch",
+            "7.9.2",
+            "9200:9200 9300:9300",
+            "discovery.type=single-node;ELASTIC_USERNAME=elastic;ELASTIC_PASSWORD=SAMPLE_PA$$WORD");
```
[the remark this answers is not in the document]
Comment (abprash) [reply src/main/java/DockerImage.java:53]: Sounds good 👍

File: src/test/java/MySQLDockerImageTest.java [hunk 4]
```diff
@@ -4,7 +4,11 @@
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.CsvSource;
 
-import java.sql.*;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
 import java.util.Arrays;
```
Comment (the-c0d3br34k3r) [inline src/test/java/MySQLDockerImageTest.java:12]: Don't include this change with this PR. I'll be refactoring it, will take care of it then.

File: src/test/java/PostgreSQLDockerImageTest.java [hunk 5]
```diff
@@ -4,7 +4,11 @@
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.CsvSource;
 
-import java.sql.*;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
```
Comment (the-c0d3br34k3r) [inline src/test/java/PostgreSQLDockerImageTest.java:11]: Don't include this change with this PR. I'll be refactoring it, will take care of it then.

File: src/test/java/SQLServerDockerImageTest.java [hunk 6]
```diff
@@ -4,7 +4,11 @@
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.CsvSource;
 
-import java.sql.*;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
```
Comment (the-c0d3br34k3r) [inline src/test/java/SQLServerDockerImageTest.java:11]: Don't include this change with this PR. I'll be refactoring it, will take care of it then.

**Review by the-c0d3br34k3r — COMMENTED** [review]
Reviewed.

File: src/test/java/MySQLDockerImageTest.java (hunk 4 above)
Comment (abprash) [reply src/test/java/MySQLDockerImageTest.java:12]: Since this is minor, I made it.

Comment (abprash) [reply src/test/java/MySQLDockerImageTest.java:12]: I think we can go ahead and get it merged.

File: src/test/java/PostgreSQLDockerImageTest.java (hunk 5 above)
Comment (abprash) [reply src/test/java/PostgreSQLDockerImageTest.java:11]: Same as above.

File: src/test/java/SQLServerDockerImageTest.java (hunk 6 above)
Comment (abprash) [reply src/test/java/SQLServerDockerImageTest.java:11]: Same as above

File: src/test/java/MySQLDockerImageTest.java (hunk 4 above)
Comment (the-c0d3br34k3r) [reply src/test/java/MySQLDockerImageTest.java:12]: Let's not do that. It's more work when we pollute commit history of files with no significance to the current change, especially when we trace code changes. Let's not set that as a practice.

File: src/test/java/PostgreSQLDockerImageTest.java (hunk 5 above)
Comment (the-c0d3br34k3r) [reply src/test/java/PostgreSQLDockerImageTest.java:11]: Let's not do that. It's more work when we pollute commit history of files with no significance to the current change, especially when we trace code changes. Let's not set that as a practice.

**Review by the-c0d3br34k3r — COMMENTED** [review]
Reviewed. Commented.

File: src/test/java/SQLServerDockerImageTest.java (hunk 6 above)
Comment (the-c0d3br34k3r) [reply src/test/java/SQLServerDockerImageTest.java:11]: Let's not do that. It's more work when we pollute commit history of files with no significance to the current change, especially when we trace code changes. Let's not set that as a practice.

File: src/test/java/PostgreSQLDockerImageTest.java (hunk 5 above)
Comment (abprash) [reply src/test/java/PostgreSQLDockerImageTest.java:11]: discussed offline. will revert

File: src/test/java/MySQLDockerImageTest.java (hunk 4 above)
Comment (abprash) [reply src/test/java/MySQLDockerImageTest.java:12]: discussed offline. will revert

File: src/test/java/SQLServerDockerImageTest.java (hunk 6 above)
Comment (abprash) [reply src/test/java/SQLServerDockerImageTest.java:11]: same as above

**Review by the-c0d3br34k3r — APPROVED** [review]
Everything looks good!
Please squash all commits into a single commit with a detailed commit message description.

**Comment by abprash** [comment]
Closing, as the history got modified in squashing, the merges from upstream master.

**Pull request closed without merging**